addons/binding/org.openhab.binding.networkhealth/src/main/java/org/openhab/binding/networkhealth/discovery/NetworkHealthDiscoveryService.java

```diff
@@ -0,0 +1,236 @@
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link NetworkHealthDiscoveryService} is responsible for discovering devices on 
+ * the current Network. It uses every Network Interface which is connect to a Network
+ * 
+ * @author Marc Mettke - Initial contribution
+ */
+
+public class NetworkHealthDiscoveryService extends AbstractDiscoveryService {
+	private final static Object lockObject = new Object();
+
+	private ScheduledFuture<?> discoveryJob;
+	private final Logger logger = LoggerFactory.getLogger(NetworkHealthDiscoveryService.class);
+
+	
+	public NetworkHealthDiscoveryService() {
+		super(SUPPORTED_THING_TYPES_UIDS, 300);
+	}
+
+	/**
+	 * Handles the whole Discovery
+	 */
+	private void discoverNetwork() {
+		TreeSet<String> interfaceIPs;
+		Queue<String> networkIPs;
+
+		logger.debug("Starting Device Discovery");
+		interfaceIPs = getInterfaceIPs();
+		networkIPs = getNetworkIPs(interfaceIPs);
+		startDisconvery(networkIPs);
+	}
+
+	/**
+	 * Gets every IPv4 Address on each Interface except the loopback
+	 * The Address format is ip/subnet
+	 * @return The collected IPv4 Addresses
+	 */
+	private TreeSet<String> getInterfaceIPs() {
+		TreeSet<String> interfaceIPs = new TreeSet<String>();
+
+		try {
+			// For each interface ...
+			for (Enumeration<NetworkInterface> en = NetworkInterface
+					.getNetworkInterfaces(); en.hasMoreElements();) {
+				NetworkInterface networkInterface = en.nextElement();
+				if (!networkInterface.isLoopback()) {
+
+					// .. and for each address ...
+					for (Iterator<InterfaceAddress> it = networkInterface
+							.getInterfaceAddresses().iterator(); it.hasNext();) {
+						
+						// ... get IP and Subnet 
+						InterfaceAddress interfaceAddress = it.next();
+						interfaceIPs.add(interfaceAddress.getAddress()
+								.getHostAddress()
+								+ "/"
+								+ interfaceAddress.getNetworkPrefixLength());
+					}
+				}
+			}
+		} catch (SocketException e) {
+		}
+
+		return interfaceIPs;
+	}
+	
+	/**
+	 * Takes the interfaceIPs and fetches every IP which can be assigned on their network
+	 * @param networkIPs The IPs which are assigned to the Network Interfaces
+	 * @return Every single IP which can be assigned on the Networks the computer is connected to
+	 */
+	private Queue<String> getNetworkIPs(TreeSet<String> interfaceIPs) {
+		Queue<String> networkIPs = new LinkedBlockingQueue<String>();
+
+		for (Iterator<String> it = interfaceIPs.iterator(); it.hasNext();) {
+			try {
+				// gets every ip which can be assigned on the given network
+				SubnetUtils utils = new SubnetUtils(it.next());
+				String[] addresses = utils.getInfo().getAllAddresses();
+				for (int i = 0; i < addresses.length; i++) {
+					networkIPs.add(addresses[i]);
+				}
+
+			} catch (Exception ex) {
+			}
+		}
+
+		return networkIPs;
+	}
+
+	public Set<ThingTypeUID> getSupportedThingTypes() {
+		return SUPPORTED_THING_TYPES_UIDS;
+	}
+
+	@Override
+	protected void startBackgroundDiscovery() {
+		new Thread(new Runnable() {

```

Comment:
A real background discovery should not only run once, but scan for new devices regularly. I am not sure though, if this makes sense - do you have any idea on the load that such a scan imposes on the system (e.g. on a Raspberry Pi)? Maybe it is better to not offer background discovery at all and only let users do a manual scan.